dubbo-remoting/dubbo-remoting-netty4/src/main/java/org/apache/dubbo/remoting/transport/netty4/NettyClient.java

```diff
@@ -78,7 +78,7 @@ protected void doOpen() throws Throwable {
 
             @Override
             protected void initChannel(Channel ch) throws Exception {
-                int heartbeatInterval = UrlUtils.getIdleTimeout(getUrl());
+                int heartbeatInterval = UrlUtils.getHeartbeat(getUrl());
```

Comment:
looks like we still should call `getIdleTimeout` instead, would you mind to check the old logic?

Reply:
Since the current server does not return a heartbeat request, if you use getIdleTimeout(3 * heartbeat), the heartbeat send period will be 3 * heartbeat. Will this be a long time?

Reply:
I think @carryxyh is right. This value tells `IdleStateHandler` the interval of sending a heartbeat package.